### DISCUSSION TITLE
mapsforge RotateMapViewer not showing map

### DISCUSSION THREAD

**Discussion by hawalter** [mapsforge/mapsforge#1298] [Q&A] [answered]
Download the current mapsforge master branch and build it with Android Studio Bumblebee | 2021.1.1 Patch 2 on macOS 12.2.1.
Install it on a Pixel 3 XL with the October 5, 2021 Android 12 patch.

Start RotateMapViewer and the Berlin map appears and the zoom buttons work. Touch the rotate button and the map disappears.

**Comment by devemux86** [comment, marked as answer]
This was an old experiment for old Androids and without proper vector rotation.

Proper map rotation exists in [VTM OpenGL map library](https://github.com/mapsforge/vtm).
(compatible with Mapsforge maps)
